flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/TaskManagerLogHandler.java

```
@@ -210,7 +210,17 @@ public void onSuccess(Object filePathOption) throws Throwable {
 							LOG.error("Displaying TaskManager log failed.", e);
 							return;
 						}
-						long fileLength = raf.length();
+						long fileLength;
+						try {
+							fileLength = raf.length();
+						} catch (IOException ioe) {
+							display(ctx, request, "Displaying TaskManager log failed.");
+							LOG.error("Displaying TaskManager log failed.", ioe);
+							if (raf != null) {
+								raf.close();
+							}
+							return;
```

let's just re-throw the exception to retain current behavior.